From febc286ab6fa43a1979afc5d046c95dae9b3e0c3 Mon Sep 17 00:00:00 2001
From: Rouven Czerwinski <r.czerwinski@pengutronix.de>
Date: Thu, 24 Mar 2022 10:03:27 +0100
Subject: [PATCH] mipi-csi2-sam: fix MEDIA MIX gasket0 dumps

Somebody forgot that the gasket regmap for the media mix already starts
at 0x60. Use an additional counter but keep the offsets within the
print, since this is at least more clear to me.

Signed-off-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>
Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
---
 drivers/staging/media/imx/imx8-mipi-csi2-sam.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-mipi-csi2-sam.c b/drivers/staging/media/imx/imx8-mipi-csi2-sam.c
index 0ce5d1c9cf52..11ff232c4a6f 100644
--- a/drivers/staging/media/imx/imx8-mipi-csi2-sam.c
+++ b/drivers/staging/media/imx/imx8-mipi-csi2-sam.c
@@ -568,13 +568,15 @@ static void dump_gasket_regs(struct csi_state *state, const char *label)
 		{ 0x64, "GPR_GASKET_0_HSIZE" },
 		{ 0x68, "GPR_GASKET_0_VSIZE" },
 	};
-	u32 i, cfg;
+	u32 i, cfg, p;
 
 	v4l2_dbg(2, debug, &state->sd, "--- %s ---\n", label);
 
+	p = 0x0;
 	for (i = 0; i < ARRAY_SIZE(registers); i++) {
-		regmap_read(state->gasket, registers[i].offset, &cfg);
+		regmap_read(state->gasket, p, &cfg);
 		v4l2_dbg(2, debug, &state->sd, "%20s[%x]: 0x%.8x\n", registers[i].name, registers[i].offset, cfg);
+		p += 0x4;
 	}
 }
 
-- 
2.39.1

